src/entry-point.tsx

```diff
@@ -41,26 +58,46 @@ export class LibraryController {
         this.reactor.raiseEvent(name, params);
     }
 
-    createLibraryByElementId(htmlElementId: string, layoutSpecsJson: any, loadedTypesJson: any) {
+    createLibraryByElementId(htmlElementId: string, layoutSpecsJson: any = null, loadedTypesJson: any = null) {
         let htmlElement: any;
         htmlElement = document.querySelector(htmlElementId) || document.getElementById(htmlElementId);
         if (!htmlElement) {
             throw new Error("Element " + htmlElementId + " is not defined");
         }
-        return ReactDOM.render(<LibraryContainer
-            libraryController={this}
-            loadedTypesJson={loadedTypesJson}
-            layoutSpecsJson={layoutSpecsJson}
-            defaultSectionString={this.DefaultSectionName}
-            miscSectionString={this.MiscSectionName} />, htmlElement);
+
+        let libraryContainer = ReactDOM.render(this.createLibraryContainer(), htmlElement);
+
+        if (loadedTypesJson && (layoutSpecsJson)) {
+            this.setLoadedTypesJson(loadedTypesJson);
+            this.setLayoutSpecsJson(layoutSpecsJson);
+            this.refreshLibraryView();
+        }
+
+        return libraryContainer;
     }
 
-    createLibraryContainer(layoutSpecsJson: any, loadedTypesJson: any) {
+    createLibraryContainer() {
         return (<LibraryContainer
             libraryController={this}
-            layoutSpecsJson={layoutSpecsJson}
-            loadedTypesJson={loadedTypesJson}
             defaultSectionString={this.DefaultSectionName}
             miscSectionString={this.MiscSectionName} />);
     }
+
+    setLoadedTypesJson(loadedTypesJson: any): void {
```

Comment:
We do need a parameter to append. Although, from Dynamo most of the time we would call it with `append: false`.